software/winrm/src/main/java/org/apache/brooklyn/util/core/internal/winrm/winrm4j/Winrm4jTool.java

```diff
@@ -75,7 +78,11 @@ public Winrm4jTool(Map<String,?> config) {
     public Winrm4jTool(ConfigBag config) {
         this.bag = checkNotNull(config, "config bag");
         host = getRequiredConfig(config, PROP_HOST);
-        port = getRequiredConfig(config, PROP_PORT);
+        port = config.get(PROP_PORT.getConfigKey()) != null
```

Comment:
This will almost never return non-null, because there is a default value - and that default value is 5985. As long as we've overridden that value with 5986 for https by the time we get here then I can see this will work.